### PR TITLE
use `tsup` for bundling instead of `webpack`

### PR DESCRIPTION
**Pull request by ckt1031**
And also fixed serval bugs:

- You can use the version from `package.json` directly, tsup bundles it for you
- Tsup automatically does CJS and ESM code transformation, so you don't have to config tsc and webpack-cli
- E2E are not using local packages, it uses production version from npm, so the tests are redundant because of that, but now it uses local's one

Changes:

```js
const { withRoutes } = require("nextjs-routes/config");
// CJS tsup doesn't support default export, so it will be changed like this

/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  typescript: {
    ignoreBuildErrors: true,
  },
};

module.exports = withRoutes()(nextConfig);
```

### CODE REVIEW

**Comment by cktsun1031** [comment]
Prettier isn't running well currently, you can format the code before merging to the main branch.

**Comment by tatethurston** [comment]
Hi @cktsun1031 tank you for contributing. Could you talk through these proposed changes a bit more?

> transformation, so you don't have to config tsc and webpack-cli

- This project uses tsc and esbuild. What advantages does moving to tsup bring? tsc will still be required for typechecking, so this this would be a substitution of esbuild for tsup, netting the same project dependencies.

> E2E are not using local packages, it uses production version from npm, so the tests are redundant because of that, but now it uses local's one

This is not true. This project uses pnpm workspaces which installs the local packages.

**Comment by tatethurston** [comment]
The change to withRoutes from a default export to a named export is a breaking change.

**Comment by cktsun1031** [comment]
Unfortunately, my machine installed the remote version instead of the local one, idk whether it's my issue or not

**Comment by cktsun1031** [comment]
The main advantage is that you can use the single cli command to bundle and export two both CJS and ESM formats, it also does bundling like solving json imports and automatically dts exporting, tsup does type checking before finishing the building.

**Comment by cktsun1031** [comment]
You can ignore 😓 this PR if you want to stay in webpack, tsc bundling instead of tsup.

File: src/config.ts [hunk 1]
```diff
@@ -5,9 +5,9 @@
 /* eslint-disable @typescript-eslint/no-unsafe-call */
 import type { NextConfig } from "next";
 import type { Configuration, WebpackPluginInstance } from "webpack";
-import { getAppDirectory, getPagesDirectory } from "./utils.js";
+import { getAppDirectory, getPagesDirectory } from "./utils";
 import { watch } from "chokidar";
-import { logger, NextJSRoutesOptions, writeNextjsRoutes } from "./core.js";
+import { logger, NextJSRoutesOptions, writeNextjsRoutes } from "./core";
```
Comment (tatethurston) [inline src/config.ts:10]: These js extensions are intentional, and required for ESM

**Pull request closed without merging**